src/network/networkd-manager.c

```diff
@@ -913,31 +913,22 @@ static int manager_save(Manager *m) {
                         const char *domainname;
                         char **domains = NULL;
 
+                        OrderedSet *target_domains = (link->network->dhcp_use_domains == DHCP_USE_DOMAINS_YES) ? route_domains : search_domains;
```

Comment:
I made a mistake in the code sample I provided. Here you should swap route_domains and search_domains, or replace DHCP_USE_DOMAINS_YES with DHCP_USE_DOMAINS_ROUTE